### PR TITLE
[4.2] Use the app identity in the FormController

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Use the app identity in the `FormController`.

~When the controller needs a current user, it must be set in the MVCFactory. This is the case on all Form based controllers.~

### Testing Instructions
- Log in on the back end
- Go to the articles manager
- Click on New
- Give the article a title
- Save the article

### Actual result BEFORE applying this Pull Request
All works.

### Expected result AFTER applying this Pull Request
All works.

### CODE REVIEW

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on cf172a9ebb05527c349ddef4ac6a84a8f6dc0ac4<br /><br />Tested successfully in Joomla 4.2.0-beta1-dev of 9 June using Wampserver 3.2.9 and PHP 8.0.15<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38011">issues.joomla.org/tracker/joomla-cms/38011</a>.</sub>

**Comment by joomdonation** [comment]
In controller, we can already access to current user object via `$this->app->getIdentity();` . Could you please explain the reason for injecting it again when controller is created using MVCFactory?

**Comment by laoneo** [comment]
YOu are right, changed the code to use the identity.

**Review by wilsonge — APPROVED** [review]

File: libraries/src/MVC/Controller/FormController.php [hunk 1]
```diff
@@ -200,9 +200,14 @@ public function add()
 	 *
 	 * @since   1.6
 	 */
-	protected function allowAdd($data = array())
+	protected function allowAdd($data = [])
 	{
-		$user = Factory::getUser();
+		$user = $this->app->getIdentity();
+
+		if (!$user)
```
Comment (joomdonation) [inline libraries/src/MVC/Controller/FormController.php]: The app identity is loaded very early in our application life cycle (at initialize phase, even before we import and trigger system plugin), so $this->app->getIdentity() will always return a `User` object, thus the check here is not needed. Same for the code which you added in `allowEdit` method.

Comment (laoneo) [reply libraries/src/MVC/Controller/FormController.php]: I'm not sure if the identity is always set in the app. @wilsonge is it?

Comment (laoneo) [reply libraries/src/MVC/Controller/FormController.php]: Can confirm, removed the check. Thanks for the hint!

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on 4a36bdebcce17e0968a9545bb22fc79a9ba5bba7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38011">issues.joomla.org/tracker/joomla-cms/38011</a>.</sub>

**Comment by joomdonation** [comment]
@laoneo Unrelated to this PR but you made similar check here https://github.com/joomla/joomla-cms/blob/4.2-dev/libraries/src/MVC/Controller/BaseController.php#L598 and here https://github.com/joomla/joomla-cms/blob/4.2-dev/libraries/src/MVC/Controller/BaseController.php#L630 in PR #37498 . The check `$this->app->getIdentity()` in these lines could be removed, too.

**Comment by laoneo** [comment]
Can you make a pr removing them?

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 4a36bdebcce17e0968a9545bb22fc79a9ba5bba7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38011">issues.joomla.org/tracker/joomla-cms/38011</a>.</sub>

**Comment by chmst** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38011">issues.joomla.org/tracker/joomla-cms/38011</a>.</sub>

**Comment by roland-d** [comment]
Thanks everybody

**Pull request merged**